File: python/file_api.py
# Copyright (C) 2020 by Algonaut USA (http://algonaut.us).
# Please see the LICENSE file that should have been included as part of this package.
# Written by Isaac de la Pena <isaac@algonaut.us>

# python file_api.py

import requests as req

sid = 'mySID'

print('Sofia File API Sample Code')

files = {"xls": open('../Sofia_Metrics_Template.xlsx', "rb")}
data = {
	"sid": sid,
	"all_dates": True
}

url = 'http://sofia.conexo.vc/xls'
# or 'http://sofia.conexo.vc/xls/%s' % sid
# or 'http://sofia.conexo.vc/xls?sid=%s' % sid

res = req.post(url, data=data, files=files)

print(res.status_code)
print(res.reason)
print(res.content)

File: node/FileApi.js
// Copyright (C) 2020 by Algonaut USA (http://algonaut.us).
// Please see the LICENSE file that should have been included as part of this package.
// Written by Isaac de la Pena <isaac@algonaut.us>

// npm install form-data

// node FileApi.js

var FormData = require('form-data');
const http = require('http')
const os = require('os')
const fs = require('fs')

const readStream = fs.createReadStream('../Sofia_Metrics_Template.xlsx');

const form = new FormData();
form.append('xls', readStream);
form.append('sid', 'mySID');
form.append('all_dates', true);

const options = {
  hostname: 'sofia.conexo.vc',
  port: 80,
  path: '/xls',
  method: 'POST',
  headers: form.getHeaders()
}

const req = http.request(options, (res) => {
  console.log(`Status Code: ${res.statusCode}`)

  res.on('data', (d) => {
    process.stdout.write(d + os.EOL)
  })
})

req.on('error', (error) => {
  console.error(error)
})

form.pipe(req);